tests/acceptance/30_generic_methods/variable_iterator.cf

```diff
@@ -0,0 +1,61 @@
+#######################################################
+#
+# Launch the /bin/date command
+#
+#######################################################
+
+bundle common acc_path
+{
+  vars:
+    "root" string => getenv("NCF_TESTS_ACCEPTANCE", 1024);
+}
+
+body common control
+{
+      inputs => { "${acc_path.root}/default.cf.sub", "${acc_path.root}/default_ncf.cf.sub", "@{ncf_inputs.default_files}" };
+      bundlesequence  => { configuration, default("${this.promise_filename}") };
+      version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "command_0"            string => "exit 0";
+    "canonified_command_0" string => canonify("${command_name}");
+}
+
+#######################################################
+
+bundle agent test
+{
+  vars:
+    "iterator1" string => 'a,b, c';
+
+  methods:
+    "ph1" usebundle => variable_iterator("prefix", "var1", "${iterator1}", ",");
+    "ph2" usebundle => variable_iterator("prefix", "var2", "${iterator1}", "\s*,\s*");
+}
+
+#######################################################
+
+bundle agent check
+{
+  classes:
+    
+    "ok_1" expression => "variable_iterator_var1_kept.!variable_iterator_var1_repaired.!variable_iterator_var1_error";
+    "ok_2" expression => "variable_iterator_var2_kept.!variable_iterator_var2_repaired.!variable_iterator_var2_error";
+    "ok_length1" expression => strcmp(length("prefix.var1"), "3");
```

Comment:
this test will work only with CFEngine 3.6
Is there something that allows to skip these test when using a 3.5 agent ?